src/panels/lovelace/hui-editor.ts

```diff
@@ -150,20 +205,31 @@ class LovelaceFullConfigEditor extends hassLocalizeLitMixin(LitElement) {
       this._saving = false;
       return;
     }
-
+    try {
+      value = lovelaceStruct(value);
+    } catch (err) {
+      alert(`Your config is not valid: ${err}`);
+      return;
+    }
     try {
       await this.lovelace!.saveConfig(value);
     } catch (err) {
       alert(`Unable to save YAML: ${err}`);
     }
+    window.onbeforeunload = null;
     this._saving = false;
     this._changed = false;
+    this._hashAdded = false;
   }
 
   private _yamlChanged() {
+    this._hash = this._hashAdded || this.textArea.value.includes("#");
```

Comment:
I think that we should skip this. I think that it's perfect to only check if a user adds a hash. That way we don't have to worry about any colors that already exist.

Reply:
The only problem is when a user pastes something but might be an edge case (it is possible to check the pasted content but might be overkill...)